src/pages/CustomerInvoicePage.js

```diff
@@ -82,30 +129,39 @@ export class CustomerInvoicePage extends GenericPage {
   };
 
   onAddMasterItems = () => {
-    this.props.runWithLoadingIndicator(() => {
-      this.props.database.write(() => {
-        this.props.transaction.addItemsFromMasterList(this.props.database);
-        this.props.database.save('Transaction', this.props.transaction);
+    const { database, transaction, runWithLoadingIndicator } = this.props;
+    runWithLoadingIndicator(() => {
+      database.write(() => {
+        transaction.addItemsFromMasterList(database);
+        database.save('Transaction', transaction);
```

Comment:
do we nee to check for vaccine added from master list here ?

Reply:
Yes, I think consolidating as I have missed a few spots is a good idea. Also ensuring that to be a vaccine doses must be set correctly (as per comment below) is also a good idea. I think in a new issue? Happy to do it on this if that's a better idea